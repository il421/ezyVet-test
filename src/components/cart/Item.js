import React from 'react';
import { formatCurrency } from '../../libs/currency.js';
import '../../styles/cart/item.css';
import '../../styles/common/button.css';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const Item = ({ id, name, price, qty, removeItem, setItemQty }) => {
  return (
    <div id={ id } className="item">
      <div className="item__name">{ name }</div>
      <div className="item__price">{ formatCurrency(price) }</div>
      <div className="item__qty-box qty-box">
        <button
          className="qty-box__dec button button--qty"
          onClick={ () => setItemQty({ id, qty: qty - 1 }) }
        >-</button>

        <input
          id={ id }
          type="number"
          value={ qty }
          onChange={ (evt) => setItemQty({ id, qty: evt.target.value }) }
        />

        <button
          className="qty-box__inc button button--qty"
          onClick={ () => setItemQty({ id, qty: qty + 1 }) }
        >+</button>
      </div>

      <button className="item__remove" onClick={ () => removeItem(id) }>
        <FontAwesomeIcon icon="times" size="2x" />
      </button>
    </div>
  );
};

export default Item;
