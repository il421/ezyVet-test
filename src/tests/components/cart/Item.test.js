import React from 'react';
import { shallow } from 'enzyme';
import Item from '../../../components/cart/Item.js';
import { cart } from '../../fixtures/data';

let wrapper, removeItem, setItemQty;

beforeEach(() => {
  removeItem = jest.fn();
  setItemQty = jest.fn();

  wrapper = shallow(
    <Item
      { ...cart[0] }
      removeItem={ removeItem }
      setItemQty={ setItemQty }
    />
  );
});

test('should render Item without crashing', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should call removeItem if click remove btn', () => {
  wrapper.find('.item__remove').simulate('click');
  expect(removeItem).toHaveBeenLastCalledWith(cart[0].id);
});

test('should call setItemQty if click increase/decrease btn', () => {
  wrapper.find('.qty-box__inc').simulate('click');
  expect(setItemQty).toHaveBeenLastCalledWith({ id: cart[0].id, qty: cart[0].qty + 1 });

  wrapper.find('.qty-box__dec').simulate('click');
  expect(setItemQty).toHaveBeenLastCalledWith({ id: cart[0].id, qty: cart[0].qty - 1 });
});

test('should call setItemQty if change qty in input box', () => {
  wrapper.find('.qty-box input').simulate('change', { target: { value: 2 } })
  expect(setItemQty).toHaveBeenLastCalledWith({ id: cart[0].id, qty: 2 });
});
