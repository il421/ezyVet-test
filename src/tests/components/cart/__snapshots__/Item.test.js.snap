// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Item without crashing 1`] = `
<div
  className="item"
  id={1}
>
  <div
    className="item__name"
  >
    Sledgehammer
  </div>
  <div
    className="item__price"
  >
    $125.75
  </div>
  <div
    className="item__qty-box qty-box"
  >
    <button
      className="qty-box__dec button button--qty"
      onClick={[Function]}
    >
      -
    </button>
    <input
      id={1}
      onChange={[Function]}
      type="number"
      value={1}
    />
    <button
      className="qty-box__inc button button--qty"
      onClick={[Function]}
    >
      +
    </button>
  </div>
  <button
    className="item__remove"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="times"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </button>
</div>
`;
